woocommerce/woocommerce-ios
# [Orders with Coupons M6] Collapsible product row component

Closes: #10833 

## Description
With this PR we add different SwiftUI components to make the product rows for Order Creation collapsable: `CollapsibleProductRowCard`, `CollapsibleProductCardPriceSummary`, and `SimplifiedProductRow`

The design is not yet as the [proposed designs](https://www.figma.com/file/FTOWxedKUyDbBdOrxRfMyI/Coupons-in-Order-Creation?type=design&node-id=14-9617&mode=design&t=gCbDARMVXEBqwarn-0), since we have to add further components and polish the details, but gets as close as possible. Some items that still needs addressing:
* The chevron for collapsing/expanding is centered due to reusing existing components, we'll duplicate this in a future PR and adapt it properly.
* There's no discounts-related functionality or design, this will be tackled on a different PR [here](https://github.com/woocommerce/woocommerce-ios/issues/10837).

![CollapsibleProductRow](https://github.com/woocommerce/woocommerce-ios/assets/3812076/32f10c63-65f4-47b2-bd2b-8b944c83c4af)


## Testing instructions
* Switch `ordersWithCouponsM6` feature flag to `true`
* Go to `Orders` > `+` > `+ Add Products` > Add different products to your order: 
  * Digital products in stock.
  * Physical products in stock where "manage stock" option is enabled, and we've given a stock quantity.

Observe that:
* Each product added to the order is collapsable, and its state is independent from other products.
* Placeholder image, product title, stock, price/quantity breakdown, and total appears and is updated when stock changes.
* Upon collapsing the view price and quantity appears, and upon expanding the view this changes to the product SKU. The border color changes as well when a product card is collapsed/expanded.
* The quantity stepper behaviour hasn't changed (this is finicky if we increase/decrease fast, but it's an underlying issue): 
  * Tapping on `+` in the quantity stepper increases product count.
  * Tapping on `-` decreases it.
  * Tapping on `-` and reaching zero removes the product from the order.
* Tapping on `Remove Product from Order` currently does nothing at the moment. We'll implement it on a [separate PR](https://github.com/woocommerce/woocommerce-ios/issues/10834).
* Taping `Create` creates the Order normally, with the expected products and quantities.
